### ISSUE TITLE
Provide return code on error

### ISSUE THREAD

**Issue by twenzel** [KirillOsenkov/SourceBrowser#89]
Please provide a return code for the HtmlGenerator when an error occurs and return only 0 when it completes successfully.

**Comment by KirillOsenkov** [comment]
Just curious what was the error that you were seeing? My philosophy was that all errors should be non-fatal and HtmlGenerator should recover from most issues that happened during generation. 

Admittedly, there's a lot that can go wrong during generation (because this is equivalent to a build: if a build fails, the generation will likely fail as well). In that case I log an error and continue.

Conceivably, I could return a non-zero exit code when any errors were logged. Would that solve the problem for you? 

Just trying to determine the level of severity at which I decide that the overall process failed and a non-zero exit code should be returned. Maybe I should also stop further generation if there was an error. Historically it was better to attempt to finish the job to at least have some website at the end, instead of nothing at all. But maybe it's not the best.

Let me know what you think.

**Comment by twenzel** [comment]
Yes, returning a non-zero exit code when any errors were logged would help/ fix my issue.

My problem was when a solution file was not found or the build fails (some internal MSBuild error). The Problem was logged correctly but my overall build continues because the HTMLGenerator was not returning a non-zero return code.
